Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+    var OVERLAY_PRESIT_URL_POSTFIX = ".png";
+    var OVERLAY_PRESIT_URL_NUM = 12;
+    var OVERLAY_PRESIT_URL_TEXT = "https://hifi-production.s3.amazonaws.com/robin/sit/sitOverlayConfirm/please-face-forward.png"
+    var overlayCheckForHover = null;
+
+    var animStateHandlerID = null;
+    var sitDownSettlePeriod = null;
+    var driveKeyPressedStart = null;
+    var sittingDown = false;
+
+    var headToHipsDistance = null;
+
+    var chairProperties = null;
+    var deadlockGate = false; // locks chair if someone has intent to sit
+
+    var testing = false;
+
+    var overlays = {
+
+        sittable: {
+            create: function () {
+                utils.setChairProperties();
+
+                // change the image based on what modality the user is in
+                var url = HMD.active
+                    ? OVERLAY_URL_SITTABLE_HMD
+                    : OVERLAY_URL_SITTABLE_DESKTOP;
+
+                overlaySittable = Overlays.addOverlay("image3d", {
+                    position: { x: 0.0, y: 0.0, z: 0.0 },
+                    rotation: Quat.multiply(chairProperties.rotation, Quat.fromVec3Degrees({ x: -90, y: 180, z: 0 })),
+                    dimensions: {
+                        x: 0.1,
+                        y: 0.1
+                    },
+                    url: url,
+                    ignoreRayIntersection: false,
+                    alpha: OVERLAY_SITTABLE_ALPHA_START,
+                    visible: true,
+                    emissive: true
+                });
+
+                var overlayDimensions = {
+                    x: 0.3,
+                    y: 0.3
+                };
+
+                var overlayPosition = Vec3.sum(chairProperties.position, { x: 0, y: 0, z: 0 });
+
+                Overlays.editOverlay(overlaySittable, {
+                    position: overlayPosition,
+                    dimensions: overlayDimensions
+                });
+
+            },
+            remove: function () {
+                if (overlaySittable !== null) {
+                    Overlays.deleteOverlay(overlaySittable);
+                    overlaySittable = null;
+                }
+            },
+            shouldShow: function () {
+                var seatPosition = Entities.getEntityProperties(entityID, ["position"]).position;
+                var distanceFromSeat = Vec3.distance(MyAvatar.position, seatPosition);
+                return distanceFromSeat < OVERLAY_SITTABLE_DISTANCE_SHOW && !utils.checkSeatForAvatar();
+            },
+            // This part is used to fade out the overlay over time
+            lerpTransparency: function () {
+                var startAlpha = OVERLAY_SITTABLE_ALPHA_START;
+                var changeAlpha = 0.01;
+                overlayIntervalTransparency = Script.setInterval(function () {
+                    startAlpha = startAlpha - changeAlpha; // My new alpha
+                    Overlays.editOverlay(overlaySittable, { alpha: startAlpha }); // Edit the existing overlay
```

Comment:
you should probably check that overlaSittable is not null, before calling editOveraly.